frontend/src/components/Search.tsx

```diff
@@ -76,6 +67,17 @@ const Search = (props: any) => {
   const [letterCodes, setLetterCodes] = useState<any[]>([]);
   const [queryString, setQueryString] = useState<string>('');
   const [numberRange, setNumberRange] = useState<number[]>([200, 400]);
+  const [names, setNames] = useState<string[]>([]);
+
+  useEffect(() => {
```

Comment:
Can we not have axios on the component but rather in an action. 
It'll make for better code reusability too

Reply:
well if we're putting it in an action, i'm assuming that it would have to alter the store (so i'm assuming you mean that you want the letter codes to be in the store). i don't see why we should do that, so i ended up calling axios in the component

Reply:
Yes, I believe it should alter the store. Why make an axios request AND a DB request _every time_ that a user goes to the Browse page?? Seems like we can save time and resources to just have it saved in the redux

+ as mentioned previously, that way other code can just reuse the values in the store too

Reply:
that's fair. in that case we face the same problem: we'll end up calling the api every time a user goes to the browse page, it doesn't really make the store act as a cache to make things faster if we're still calling the api each time

but yea having other parts of the app reuse those values is a good idea, but it doesn't change the fact that we're not making api calls more efficient  (making fewer api calls)

Reply:
We can make either
- the backend avoid making the DB call if it has the value cached there or 
- the action avoid making the API call if we find that the store has the values

I prefer the second, and will still be reliable since our DB data rarely changes

Reply:
Seconded on the check, it doesn't cost too much to have the API call in my opinion but reducing an unnecessary call would be nice too

Reply:
yea i'm not sure if the store is supposed to be used as a cache honestly, but i'm most likely wrong

anyway sure, i'll transfer it over to the store but i won't implement the api optimization yet cuz ngl i don't really know how, seems kind of unintuitive to check the store for values when the whole point of the store is to contain dynamic data that is typically used to update components

Reply:
Since we know the scale of the list of strings, I think we can safely use it. 
And it kind of can be used as a cache, at least from my experience looking at different ways libraries use the store.

Not sure what you mean by API optimization in this context; I thought it would just be:
```
if not in store
 call API
else
  use store
```